### ISSUE TITLE
Libvirt provider: Molecule destroy does not work well when provider parameters are wrong

### ISSUE THREAD

**Issue by oserde** [metacloud/molecule#123]
Molecule destroy does not work well when libvirt provider parameters are wrong or not supported by the machine. 

For instance, when I specify "graphics_type: gtk", i get the following error:

> Error while creating domain: Error saving the server: Call to virDomainDefineXML failed: unsupported configuration: unknown graphics device type 'gtk'

Then i execute `molecule destroy` and nothing happens. If I try to execute molecule again(`molecule test`, or` molecule create`), I get the following error:

> Volume for domain is already created. Please run 'vagrant destroy' first.

So while using libvirt and setting wrong or unsupported parameters, I have to remove the volume manually. I do not know whether this happens while using vagrant.

I have observed this section in `comands.py`:

```
class Create(AbstractCommand):
    def execute(self):
        """
        Creates all instances defined in molecule.yml.
        Creates all template files used by molecule, vagrant, ansible-playbook.

        :return: None
        """
        if self.static:
            self.disabled('create')

        self.molecule._create_templates()
        try:
            self.molecule._provisioner.up(no_provision=True)
            self.molecule._state['created'] = True
            self.molecule._write_state_file()
        except CalledProcessError as e:
            print('ERROR: {}'.format(e))
            sys.exit(e.returncode)
```
I would suggest to introduce a vagrant destroy call inside the except block.

**Comment by retr0h** [comment]
@oserde Is this still an issue with the current libvirt work that was merged in #119?

**Comment by retr0h** [comment]
Closing.  Please re-open if still an issue.
